essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/EssentialsMod.kt

```diff
@@ -24,17 +24,23 @@
 
 package io.github.nyliummc.essentials.api
 
+import com.google.inject.Inject
 import net.minecraft.server.MinecraftServer
+import java.lang.IllegalArgumentException
 
 interface EssentialsMod {
     val server: MinecraftServer
     val isClient: Boolean
     val registry: EssentialsRegistry
     val database: EssentialsDatabase
 
-    // TODO: Cleaner way to do this
     companion object {
+        @field:Inject
+        private var instance: EssentialsMod? = null
```

Comment:
rename to `implementation` or similar, that way `.instance` can be used instead of `.getInstance()`